fmf.spec

```diff
@@ -45,35 +38,44 @@ This package contains the Python 3 module.
 
 
 %prep
-%autosetup
+%autosetup -p1 -n fmf-%{version}
+
+
+%generate_buildrequires
+%pyproject_buildrequires -x tests %{?epel:-w}
 
 
 %build
-%py3_build
+%pyproject_wheel
+cp docs/header.txt man.rst
+tail -n+7 README.rst >> man.rst
+rst2man man.rst > fmf.1
 
 
 %install
-%py3_install
+%pyproject_install
+%pyproject_save_files fmf
+
 mkdir -p %{buildroot}%{_mandir}/man1
 install -pm 644 fmf.1* %{buildroot}%{_mandir}/man1
 
 
 %check
-%{__python3} -m pytest -vv -c tests/unit/pytest.ini -m 'not web'
-
+%pytest -vv -m 'not web'
 
-%{!?_licensedir:%global license %%doc}
 
 %files
 %{_mandir}/man1/*
 %{_bindir}/%{name}
 %doc README.rst examples
 %license LICENSE
 
-%files -n python%{python3_pkgversion}-%{name}
-%{python3_sitelib}/%{name}/
-%{python3_sitelib}/%{name}-*.egg-info
+%files -n python3-fmf -f %{pyproject_files}
+# Epel9 does not tag the license file in pyproject_files as a license. Manually install it in this case
```

Comment:
hmm, tmt doesn't have this workaround and I can see the file in the el9 rpm. In:
`/usr/lib/python3.9/site-packages/tmt-1.35.0.dist-info/licenses/`

Reply:
ah, I think we need `-f %{pyproject_files}`

Reply:
This is actually quite tricky, apparently *only* hatchling backend uses this approach right now. Still waiting on [PEP639](https://discuss.python.org/t/pep-639-round-3-improving-license-clarity-with-better-package-metadata/53020) to conclude. But Epel9 workaround wise, I am not sure if the issues was that the macros are outdated or whatnots. I need to check back on https://github.com/packit/packit/pull/1913#discussion_r1163956946 and refresh my thoughts on this.

Reply:
Fair point. Looking at PEP639, we might want to reconsider the way we have it here. I'd lean towards not doing conditionals in spec file if possible though.  

@LecrisUT of course you're involved in the PEP discussion :D 
![image](https://github.com/user-attachments/assets/6435e0f0-a1bb-499e-a631-19165fddbe62)